src/templates/Saved/index.tsx

```diff
@@ -3,26 +3,30 @@ import CategoryLayout from '@components/CategoryLayout';
 import { CardContainer, Content, Title } from './styles';
 import CategoryCard from '@components/CategoryCard';
 import DeleteComponent from '@components/DeleteComponent';
+import { useCategories } from '@hooks/useCategories';
 
 interface IMyCoursesTemplateProps {
-  categorys: ICategory[];
+  categories: ICategory[];
 }
 
-export function MyCoursesTemplate({ categorys }: IMyCoursesTemplateProps) {
+export function SavedCategoriesTemplate({ categories }: IMyCoursesTemplateProps) {
+
+  const { deleteCategory} = useCategories()
+
   return (
     <CategoryLayout>
       <Title>
         <h1>Cursos salvos</h1>
       </Title>
       <Content>
-        {categorys.map((category) => {
+        {categories.map((category) => {
           return (
             <CardContainer key={category.id}>
               <CategoryCard
                 img={category.img}
                 lessons={category.lessons}
                 title={category.name}
-                deleteIcon={<DeleteComponent/>}
+                deleteIcon={<DeleteComponent handleDelete={()=> deleteCategory(category.id)}/>}
```

Comment:
Aqui já vale aquela otimização né? :)

Reply:
tipo, eu posso fazer.. mas não to entendendo muito..
colocar a arrow assim parece que é pratica do react, recomendado por eles mesmos. Eu andei lendo e assuntando por aqui, e realmente acontece o que vc disse, mas parece q o react é performático em outros sentidos, o que compensaria fazendo com que não faça mta diferença esse consumo de memória.
Mas mesmo seguindo o que nós conversamos, eu tinha entendido que vc disse que essa otimização seria para casos onde a iteração, porque eu estaria salvando a mesma funçao varias vezes.. nesse caso, como no outro acima, é um componente só

Reply:
Mesmo comentário ali de cima, se esse componente não estiver dentro de algum map em outro lugar, tudo bem, segue o baile, mas faz um double-check só.